src/common/transformations/tests/common_optimizations/matmul_multiply_fusion.cpp

```diff
@@ -77,24 +78,71 @@ TEST_F(TransformationTestsF, MatMulMultiplyFusionConstantTransposedWeightsNonSca
 }
 
 TEST_F(TransformationTestsF, MatMulMultiplyFusionNonConstantTransposedWeightsNonScalarConstant) {
+    {
+        auto data = std::make_shared<opset8::Parameter>(element::f32, Shape{2, 3});
+        auto weights = opset8::Constant::create(element::f32, Shape{2, 3}, {1, 1, 1, 1, 1, 1});
+        auto matmul = std::make_shared<opset8::MatMul>(data, weights, false, true);
+        auto mul_const = opset8::Constant::create(element::f32, Shape{1, 2}, {1, 1});
+        auto mul = std::make_shared<opset8::Multiply>(matmul, mul_const);
+        model = std::make_shared<Model>(NodeVector{mul}, ParameterVector{data});
+
+        manager.register_pass<ov::pass::MatMulMultiplyFusion>();
+        ov::pass::Manager a;
+        a.register_pass<ov::pass::MatMulMultiplyFusion>();
+        a.run_passes(model);
+    }
+
+    {
+        auto data = std::make_shared<opset8::Parameter>(element::f32, Shape{2, 3});
+        auto weights = opset8::Constant::create(element::f32, Shape{2, 3}, {1, 1, 1, 1, 1, 1});
+        auto matmul = std::make_shared<opset8::MatMul>(data, weights, false, true);
+        model_ref = std::make_shared<Model>(NodeVector{matmul}, ParameterVector{data});
+    }
+    comparator.enable(FunctionsComparator::CmpValues::CONST_VALUES);
+}
+
+TEST_F(TransformationTestsF, MatMulMultiplyFusionConstantWeightsMarkedToKeepSrcPrecision) {
+    {
+        auto data = std::make_shared<opset8::Parameter>(element::f32, Shape{1, 2, 4, 3});
+        auto weights = opset8::Constant::create(element::f32, Shape{3, 2}, {1, 2, 3, 4, 5, 6});
+        mark_as_dequantization_node(weights);
```

Comment:
Only Subtract/Multiply nodes can be marked with dequantization attribute. Can we create weights in i8 precision, add convert (i8->f32), and multiply which is marked as dequantization node? This would make the test case more similar with real situation.

Reply:
Please check

Reply:
Sorry, I wasn't clear enough. I meant that we need to create **second** mul on weights and mark it with dequantization attribute. First mul (whose parent is matmul) should not be marked

Reply:
@v-Golubev Please check